centinel/primitives/traceroute.py

```diff
@@ -28,6 +28,10 @@ def traceroute(domain, method="udp", cmd_arguments=[], external=None):
     # the method specified by the function parameter here will
     # over-ride the ones given in cmd_arguments because
     # traceroute will use the last one in the argument list.
+
+    if cmd_arguments is None:
```

Comment:
Defaulting to an empty array will make it a static array that keeps getting appended to everytime this function is called, which makes it grow with duplicate values (an obvious memory leak). This way we can limit the scope to inside of the function and make sure it's deleted every time the function returns.